pkg/ruler/ruler.go

```diff
@@ -187,26 +188,50 @@ func NewRuler(cfg Config, engine *promql.Engine, queryable promStorage.Queryable
 		logger:       logger,
 	}
 
+	if cfg.EnableSharding {
+		ringStore, err := kv.NewClient(cfg.Ring.KVStore, ring.GetCodec())
+		if err != nil {
+			return nil, errors.Wrap(err, "create KV store client")
+		}
+
+		if enableSharding(ruler, ringStore) != nil {
```

Comment:
You're not catching the correct error here:

```suggestion
		if err = enableSharding(ruler, ringStore); err != nil {
```

Reply:
Good catch!